tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseEvaluation.scala

```diff
@@ -0,0 +1,148 @@
+package org.allenai.nlpstack.parse.poly.eval
+
+import org.allenai.nlpstack.parse.poly.polyparser._
+
+object ParseEvaluation {
+
+  /** Runs a specified scoring function over all parses in the parse source, and sums the results.
+    *
+    * Each scoring function returns a ratio (i.e. numerator/denominator) for each parse in
+    * the parse source. The sum of these ratios is returned.
+    *
+    * @param scoringFunction the scoring function to apply to the parses
+    * @param parseSource the source of parses
+    * @return the sum of the ratios produced by calling the scoring function on each parse of the
+    * parse source
+    */
+  def scoreParseSource(
+    scoringFunction: ParseScore,
+    parseSource: PolytreeParseSource
+  ): (Int, Int) = {
+
+    (parseSource.parseIterator map { candidateParse =>
+      scoringFunction.getRatio(candidateParse)
+    }) reduce { (x, y) => (x._1 + y._1, x._2 + y._2) }
```

Comment:
Was this really intended? You're adding the numerators and the denominators separately to produce the sum of two fractions-- that wouldn't produce the right result. 1/2 + 1/2 is not 2/4 (which is 1/2). May be I'm missing something?